flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileReaderOperator.java

```diff
@@ -449,7 +450,7 @@ private void cleanUp() {
 			try {
 				r.run();
 			} catch (Exception e) {
-				firstException = ExceptionUtils.firstOrSuppressed(firstException, e);
+				firstException = ExceptionUtils.firstOrSuppressed(e, firstException);
```

Comment:
Could we add some simple test for this bug?

Reply:
It won't be reliable - the only difference will be in error message and stacktrace.

Reply:
> the only difference will be in error message and stacktrace.

Isn't that's the whole point of using `firstOrSuppressed` - to have the correct exception message?

Reply:
I don't think we can have 100% coverage.
Here the fix is trivial, it's check is trivial, the importance is low, the test is relatively difficult (operator init with all the `Task` required) and un-reliable.

Reply:
Potential test doesn't look difficult:

```
	@Test(expected = ExpectedTestException.class)
	public void test() throws Exception {
  		// duplicated setup code
  		TextInputFormat format = new TextInputFormat(new Path(testBasePath)) {
			@Override
			public void close() {
				throw new ExpectedTestException();
			}
		};
        // more duplicated code that could be deduplicated
       	try (OneInputStreamOperatorTestHarness<TimestampedFileInputSplit, String> tester = ...) {
			tester.open();
		}
}
```
It's also reliable, assuming we get rid of the `FlinkRuntimeException` wrapping, which looks unnecessary.

That's ~10 lines of simple code after deduplication?

Reply:
It requires some more configuration like time characteristic and types.
I'm not saying it's not possible or difficult, I'm just saying it doesn't worth it.